app/assets/stylesheets/layout.scss

```diff
@@ -449,6 +449,40 @@ a {
   margin-top: 0;
 }
 
+.admin {
+
+  .side-menu-and-content {
+    min-height: calc(100vh - 110px);
```

Comment:
Just curious: where are these numbers (110px here and 181px for the dashboard) coming from? :thinking:

On my machine this makes a scrollbar appear in the browser's window. Maybe it's just a me, but I've always found it a usability issue: the scrollbar appears and I feel it's asking me to scroll and check the content below... but then, there's nothing below :sweat_smile:.

![Scrollbar appearing when there is no content below](https://user-images.githubusercontent.com/35156/80207686-e6fa4380-862e-11ea-9e63-4f255f5a4011.png)

Do you think it's possible to use a more precise unit, or will it always depend on the browser and screen resolution?

By the way, would it be easier to accomplish this goal if we had less `<div>` tags in that part of the code? I made an experiment a few months ago, (see javierm#12), but as we mentioned back then, I wanted to know your opinion before continuing working on it.

Reply:
The numbers come from headers height: **110px** from admin header and **181px** from the front header.

Those pixels are absolute, I mean, they couldn't be more accurate. 😛 The problem of showing or not the scroll bar I think is complicated because it depends a lot on the screen resolution, I have tried in my laptop and in two external screens and it doesn't appear to me. Anyway, I personally prefer a scroll bar to "nothing", which appears on many websites and I can understand that it is due to the size of my screen instead of blank space that gives the impression that the layout is broken.

I have looked at https://github.com/javierm/consul/pull/12 the PR but I do not see the point. If I understand correctly, to simplify the HTML, JS custom is added. This seems to me more complicated to maintain since CONSUL uses a framework, so... I don't see sense to create "custom" files to add a function that the Foundation framework already does by default. The HTML may seem more complicated but you can always consult the official documentation, if you add JS custom it seems more complicated to me for someone who wants to make a modification.

I don't see a simple solution since it will always depend on the browser and screen resolution. Also, as these changes are related to fixing minor design details administration panel I don't see the point of creating a separate PR. 🙏

Reply:
> Those pixels are absolute, I mean, they couldn't be more accurate. :stuck_out_tongue:

Maybe we're talking about different things :sweat_smile:. I mean, the headers do not always have that height. Here are screenshots of firefox and chrome on my machine:

![Header in Firefox: 124.65px](https://user-images.githubusercontent.com/35156/80703834-ce34d680-8ae3-11ea-8cda-6a993b98c085.png)

![Header in Chrome: 117px](https://user-images.githubusercontent.com/35156/80703886-e73d8780-8ae3-11ea-9da8-4858ee29f122.png)

Not to mention they also change depending on the user's font size :relieved:.

It seems to me using relative units wouldn't solve the issue either :thinking:.

Do you think it's feasible to use `flex`, which would solve these problems on every browser?

> if you add JS custom it seems more complicated to me for someone who wants to make a modification.

Fair point. It would be great to know what's harder to maintain: a custom view which will have to be changed every time the original view changes or a custom JavaScript file changing the behavior defined in the original JavaScript file :thinking:.

> I personally prefer a scroll bar to "nothing"

We agree to disagree :smile:. Unfortunately we can't do usability tests to check which option is worse :sweat_smile:.

Reply:
I've updated javierm#12 so it doesn't use JavaScript anymore.

The behaviour is not the same as it is now, though :relieved:. Let me know what you think :wink:.

Reply:
EDIT: the behaviour is now similar to the `master` branch :smile:.

Reply:
I've updated the pull request once again. Now it doesn't use JavaScript and the back button isn't broken either. It uses a (hidden) checkbox to toggle the menu, which IMHO semantically makes more sense than a link.

Reply:
One more update. I've added a commit which solves the blank space on the bottom and doesn't add a scrollbar. It seems to be really simple after taking all the `off-canvas` div tags.

Note so far I've only done changes in the admin section; the dashboard layout is probably broken right now.

Reply:
@decabeza I've improved the code a bit and tested the cases I could think of. Feel free to check pull requests #4005 and #4006 :wink: (only if you'd like to do so, of course!)

Regarding this pull request, I was thinking maybe we could remove the commits removing the mixin (already included in #4005) and fixing the blank space (solved by #4006) :thinking:. What do you think?

Reply:
Proceeding as mentioned, as #4005 has already been merged.